### ISSUE TITLE
[Legitimate Site Blocked] behodler.io

### ISSUE THREAD

**Issue by gititGoro** [MetaMask/eth-phishing-detect#58043]
https://behodler.io/

I'm not sure if this is because of a recent exploit attack but the team took remedial steps to lock withdraws. The dapp itself is completely legitimate and the code in the github repo matches the deployed code.

This block message seems to suggest the product itself is a scam which is not the case.

**Comment by AlexHerman1** [comment]
can you provide more details on the exploit please?

**Comment by gititGoro** [comment]
> can you provide more details on the exploit please?

Hi there, sure.

We've just released a post mortem report
https://github.com/Behodler/behodler2/blob/post-mortem-foundry/ExploitReport.md

The TL;DR is that the liquidity token for this AMM has an arithmetic assymetry in the mint and burn functions because of an arithmetic rounding bug.

The attacker was able to mint a little bit more than what a burn required and so performing this action in rapid succession was able to inflate the liquidity token. The token itself trades on Uniswap, leaving the attacker the option to either redeem liquidity or to sell on Uniswap. The latter was chosen.

Because the AMM is a single pool of liquidity, rather than pairs, it has safety parameters in place. One of them is a limit on withdrawals per tx. We took action in setting this to 0% after the exploit so that the remaining liquidity is locked until we provide a fix for the bug.
